cpp/src/replace/clamp.cu

```diff
@@ -90,18 +90,16 @@ std::unique_ptr<cudf::column> clamp_string_column(strings_column_view const& inp
   // build offset column
   auto offsets_transformer = [lo_itr, hi_itr, lo_replace_itr, hi_replace_itr] __device__(
                                string_view element, bool is_valid = true) {
-    const auto d_lo         = (*lo_itr).first;
-    const auto d_hi         = (*hi_itr).first;
-    const auto d_lo_replace = (*lo_replace_itr).first;
-    const auto d_hi_replace = (*hi_replace_itr).first;
-    const auto lo_valid     = (*lo_itr).second;
-    const auto hi_valid     = (*hi_itr).second;
+    const auto d_lo         = (*lo_itr).value_or(element);
+    const auto d_hi         = (*hi_itr).value_or(element);
```

Comment:
Maybe just a personal preference:
```suggestion
    const auto d_lo         = lo_itr->value_or(element);
    const auto d_hi         = hi_itr->value_or(element);
```

Reply:
I agree that I prefer this style change when comparing the two.

Reply:
I had to revert these changes as thrust iterator doesn't support `->`.

https://github.com/NVIDIA/thrust/blob/main/thrust/iterator/iterator_facade.h#L240